server/application/application.go

```diff
@@ -144,6 +153,69 @@ func (s *Server) Watch(q *ApplicationQuery, ws ApplicationService_WatchServer) e
 	return nil
 }
 
+// validateApp will ensure:
+// * the git repository is accessible
```

Comment:
I do agree in principle, but I generally prefer to do something like that on an as-needed basis to avoid function sprawl. For now the only callers that need this are application add and update, and they just want to validate the app as a whole, instead of validating different aspects.